inc/Engine/Optimization/LazyRenderContent/Frontend/Processor/SimpleHtmlDom.php

```diff
@@ -30,6 +30,26 @@ class SimpleHtmlDom implements ProcessorInterface {
 	 */
 	private $max_hashes;
 
+	/**
+	 * Array of patterns to exclude from hash injection.
+	 *
+	 * @since 3.17
+	 *
+	 * @var array
+	 */
+	private $exclusions;
+
+	/**
+	 * Sets the exclusions list
+	 *
+	 * @param string[] $exclusions The list of patterns to exclude from hash injection.
+	 *
+	 * @return void
+	 */
+	public function set_exclusions( $exclusions ): void {
+		$this->exclusions = $exclusions;
+	}
```

Comment:
I was thinking about having an abstract class to have those common functionalities instead of repeating this block of code across all processors.

I mean having `AbstractProcessor` which implements the interface and then extend all processors from it.

I know that we may remove other processors but this is a tiny step that will make the code more organized and maintainable.

what do u think?

Reply:
I don't see much value adding an abstract class that might be removed later, for something small too. We could revisit that later if common code between the implementations increases.